CMakeLists.txt

```diff
@@ -83,6 +94,11 @@ if(CREATE_TEST_TARGETS AND NOT WIN32)
 	include(gtest)
 endif()
 
+if (BUILD_SHARED_LIBS)
+	string(REGEX MATCH "^[0-9]+\\.[0-9]+\\.[0-9]+" FALCOSECURITY_SHARED_LIBS_VERSION ${FALCOSECURITY_LIBS_VERSION} )
```

Comment:
I'm wondering, do these work when `FALCOSECURITY_LIBS_VERSION` is a commit hash instead of a git tag version?

Reply:
It can be any valid git ref (a tag, a commit id, a branch name). We have to support that, since it's very handy when developing/debugging

Reply:
Thanks for pointing this out. I moved the shared version check to its own function and made it more robust.